src/agent/discovery_proxy.cpp

```diff
@@ -222,12 +222,18 @@ void DiscoveryProxy::OnHostDiscovered(const std::string &
                                       const Mdns::Publisher::DiscoveredHostInfo &aHostInfo)
 {
     otDnssdHostInfo     hostInfo;
-    const otDnssdQuery *query = nullptr;
+    const otDnssdQuery *query            = nullptr;
+    std::string         resolvedHostName = aHostInfo.mHostName;
 
     otbrLogInfo("host discovered: %s hostname %s addresses %zu", aHostName.c_str(), aHostInfo.mHostName.c_str(),
                 aHostInfo.mAddresses.size());
 
-    CheckHostnameSanity(aHostInfo.mHostName);
+    if (resolvedHostName.empty())
```

Comment:
In what cases will the `resolvedHostName` be empty?

Reply:
This is possible when mDNS found no IPv6 address for this host name.
In such case, I think it's meaningful to notify the Discovery Proxy with empty hostname and zero addresses.

Thoughts?